practices/appeng/adrs/0006-use-graphql.md

```diff
@@ -0,0 +1,83 @@
+
+For the API transport layer in particular, this is a decision where there are many different ways to accomplish the same goal, which is essentially to provide a medium for the app to make requests of the API. GraphQL is good enough and the benefits of standardizing outweigh any of the benefits of alternatives.
+
+## When to Not Implement This Decision
+
+- If you need to work with multi-part uploads you will probably need to do that outside of graphQL at the very least.
+- If your project requires very high network performance (or very low network traffic) other tools will give you more control over message size
+- If you need to build an API for external consumption, GraphQL might be too big of an ask for other teams to figure out
+
+## Alternatives Considered
+
+### Do Nothing
+
+Keep this off the list of truss recommendations, allow teams to continue to choose what transport protocol to use from scratch every time
+
+- `+` So easy
+- `+` There are lots of considerations that go into picking a transport protocol, not having a rec would force teams to consider the decision carefully every time
+- `-` With different teams doing their own thing, it becomes much harder for us to share tooling and experience when starting a new project. If we pick a default, it opens the door to better collaboration as a contracting firm
+
+### [GraphQL](https://graphql.org)
+
+GraphQL is a strictly specified protocol built on top of HTTP that exposes your API as a graph of related entities, providing mechanisms for retrieving related entities along with any request.
+
+- `+` Strictly specifies the types of everything that can be sent over the wire. It specifies how arguments are passed and parsed and specifies how to model relationships between entities. This saves everyone time, those questions are already answered.
+- `+` Has a “modern” type system with union types and optionals
+- `+` Has a sophisticated system for requesting related entities from any request. This allows the client to fetch what it needs without having to make N+1 HTTP requests, or forcing the server to build a custom endpoint
+- `+` [Apollo Client](https://www.apollographql.com/docs/react/) is an excellent React library for doing GraphQL, it makes it easy for different components to trigger the requests they need and handles caching
+- `+` Uses a schema to define the whole API, this can be used as the basis for generating web-app and server code. Having a schema makes it clear where the single source of truth for the API is, and is useful as part of a process for making changes.
+- `+` Custom types allow you to parse types into custom types on the server, offloading validating that the request matches the spec to the framework.
+- `+` Docs are good
+- `-` GraphQL is not on rails, setting up graphql across your stack for the first time requires some plumbing work
```

Comment:
Would like to see this fleshed out more. What kinds of work is involved in set up, what models are there?